spec/services/dates/validate_selected_weekly_date_spec.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Dates::ValidateSelectedWeeklyDate do
+  subject(:service) { described_class.new(params: date_partials) }
+
+  context 'when date is in correct format and is in range' do
+    year = Time.current.year
+    month = Time.current.month
+    day = Time.current.day
```

Comment:
move it to let variables and use it in each context

Reply:
Like Sergi advised, change to let statements, and only overide them in each context and do not overide `date_partials` then.
```suggestion
    let(:year) { Time.current.year }
    let(:month) { Time.current.month }
    let(:day) { Time.current.day }
```